.github/CONTRIBUTING.md

```diff
@@ -9,8 +9,92 @@ If you wish to submit more complex changes, please sync with a core developer fi
 This will help ensure those changes are in line with the general philosophy of the project
 and enable you to get some early feedback.
 
-See the [contributing guide](https://docs.celo.org/community/contributing) for details on how to participate.
+See the [community contribution guide](https://docs.celo.org/community/contributing) for details on how to participate.
+
+## Directory Structure
+
+<pre>
+├── <a href="./docs">docs</a>: Documentation on how SocialConnect works and how to use it
+├── <a href="./kubernetes-deployments">kubernetes-deployments</a>: YAML config files and instructions for ODIS deployment
+├── <a href="./apps">apps</a>: Contains deployed componentes of Oblivious Decentralized Identifier Service (ODIS) for SocialConnect
+│   ├── <a href="./apps/combiner">combiner</a>: Orchestrates distributed BLS threshold signing with the set of ODIS signers - requests and combines partial signatures
+│   ├── <a href="./apps/monitor">monitor</a>: Monitoriing service that sends health checks to deployed ODIS instances. Also contains code for load testing
+│   ├── <a href="./apps/signer">signer</a>: Generates unique partial signatures for blinded messages
+├── <a href="./packages">packages</a>: Contains all published SocialConnect components
+│   ├── <a href="./packages/common">common</a>: Contains common logic for ODIS, including API schemas
+│   ├── <a href="./packages/encrypted-backup">encrypted-backup</a>: PEAR account recovery SDK, powered by ODIS
+│   ├── <a href="./packages/identity">identity</a>: SDK for using SocialConnect
+│   ├── <a href="./packages/odis-identifiers">odis-identifiers</a>: Contains identifier prefixes and hashing functions for ODIS
+├── <a href="./scripts">scripts</a>: Misc. deployment and release scripts
+</pre>
 
 ## Dev Setup
 
-We use yarn berry. You may need to run `corepack enable`. This doesnt setup yarn itself but is a node.js feature for enabling package managers.
+### Pre-requisites
+
+* [Git](https://git-scm.com/downloads)
+* [NodeJS](https://nodejs.org/en/download/)
+* [Node Version Manager](https://github.com/nvm-sh/nvm)
+
+### Setup
+
+Clone the repository and open it:
+
+```bash
+git clone git@github.com:celo-org/social-connect.git
+cd social-connect
+```
+
+#### Install the Correct Version of NodeJS
+
+Install the correct node version with [nvm](https://github.com/nvm-sh/nvm)
+
+```bash
+nvm use
+```
+
+#### Install node modules and build with yarn
+
+We use yarn berry. You may need to run `corepack enable`. This doesn't setup yarn itself but is a node.js feature for enabling package managers.
+
+If you don't have yarn installed, run
+
+```bash
+npm i -g yarn
```

Comment:
actually im fairly certain that you dont need to do this.

Reply:
as it would be installing yarn 1

Reply:
Are you suggesting I remove this step entirely or edit it. If the latter, could you please specify the command you would like here?

Reply:
I guess I don't see how yarn could be installed automatically if it hasn't been previously. I also would think that the latest version would be installed by this command. Could you fill me in?

Reply:
yeah so i was a bit confused about this at first too. but `corepack enable` gives the `yarn` command and then it seems full yarn version that is needed is installed when the project is being installed 

https://yarnpkg.com/getting-started/install

only version 1 of yarn is on npm

Reply:
got it, thanks!